packages/perseus-core/src/utils/get-save-warnings-for-item.ts

```diff
@@ -0,0 +1,74 @@
+import type {PerseusItem, PerseusWidgetsMap, RadioWidget} from "../data-schema";
+
+/* ------- Main save warnings function for the whole item ------- */
+
+// TODO(LEMS-3643): Remove the "WORK IN PROGRESS" comment below once all
+// widgets have been migrated and the function is ready for use.
+/**
+ * WORK IN PROGRESS: Get the save warnings (blocking issues) for a given PerseusItem.
+ *
+ * @param item - The PerseusItem that will be fully parsed and checked
+ *                for blocking changes that should prevent the item from being
+ *                saved/published.
+ * @returns An array of strings, each representing a warning that should prevent
+ *          the item from being saved/published.
+ */
+export function getSaveWarningsForItem(item: PerseusItem): Array<string> {
+    const allSaveWarnings: Array<string> = [];
+    const widgets: PerseusWidgetsMap = item.question.widgets;
+    const content = item.question.content;
+
+    // Only check widgets that are actually referenced in the content
+    // (widgets can remain in the widgets object after being deleted from content)
+    for (const [widgetId, widget] of Object.entries(widgets)) {
+        if (!content.includes(widgetId)) {
+            continue;
+        }
+
+        switch (widget.type) {
+            case "radio":
```

Comment:
Could we just create a dictionary of widget type to validator function here?  I kinda feel like the pattern we should use is to create a new function on the [`WidgetLogic`](https://github.com/Khan/perseus/blob/843d66257ceb32b7c3d3eea4dea47dfda58e7945/packages/perseus-core/src/widgets/logic-export.types.ts#L57) type and then look it up the way we do the various options defined there (such as [`getCurrentVersion()`](https://github.com/Khan/perseus/blob/47ba785e06a85b7592c55da19ffa3656bea710b7/packages/perseus-core/src/widgets/core-widget-registry.ts#L57) does).